packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx

```diff
@@ -4,24 +4,49 @@ import { lock } from '@wordpress/icons';
 import clsx from 'clsx';
 import { useTranslate } from 'i18n-calypso';
 
-interface DomainsTableSSLCellProps {
+interface DomainsTableSslCellProps {
 	domainManagementLink: string;
 	sslStatus: DomainData[ 'ssl_status' ];
+	hasWpcomManagedSslCert?: DomainData[ 'has_wpcom_managed_ssl_cert' ];
 }
 
-export default function DomainsTableSSLCell( {
+export default function DomainsTableSslCell( {
 	domainManagementLink,
 	sslStatus,
-}: DomainsTableSSLCellProps ) {
+	hasWpcomManagedSslCert,
+}: DomainsTableSslCellProps ) {
 	const translate = useTranslate();
+	// WordPress.com managed subdomains (e.g. *.wordpress.com, *.wpcomstaging.com, etc.)
+	// are covered by a wildcard SSL cert, so we display them as 'Active'.
+	const isActiveSsl = sslStatus === 'active' || hasWpcomManagedSslCert;
+	const isPendingSsl = sslStatus === 'pending' || sslStatus === 'newly_registered';
+	const domainHasSsl = sslStatus || hasWpcomManagedSslCert;
 
-	const isPendingSSL = sslStatus === 'pending' || sslStatus === 'newly_registered';
+	const Element = ( { children, ...props }: { children: React.ReactNode; className: string } ) => {
+		if ( sslStatus ) {
+			return (
+				<a
+					href={ `${ domainManagementLink }?ssl-open=true` }
+					onClick={ ( event ) => event.stopPropagation() }
+					{ ...props }
+				>
+					{ children }
+				</a>
+			);
+		}
+
+		if ( hasWpcomManagedSslCert ) {
+			return <span { ...props }>{ children }</span>;
+		}
 
-	const getSSLStatusText = () => {
-		if ( sslStatus === 'active' ) {
+		return '-';
+	};
```

Comment:
Registered domains with their own certs render a link to view domain and certificate details. Because that isn't an option for subdomains, we'll render a `span` instead.

With this new logic, the `-` we rendered previously should never be reached, but I've left it in as a fallback. If that return value ever is reached, it'll look better if it isn't blank and matches the dashes shown elsewhere on the table.